api/v1alpha1/bundle_webhook.go

```diff
@@ -62,9 +59,16 @@ func (r *Bundle) ValidateDelete() error {
 	return nil
 }
 
-func checkNameLength(r *Bundle) error {
-	if len(r.Name) > maxNameLength {
-		return fmt.Errorf("bundle name %s is too long: maximum allowed name length is %d", r.GetName(), maxNameLength)
+func checkBundleSource(r *Bundle) error {
```

Comment:
Were you able to try the union type validations described here?

Wondering if this PR is a stepping stone before that or if there's an issue using those in the CRD schema?

Reply:
Yes, the union type validation works.  e2e testing is verifying that the only one of the union items can be specified.

Reply:
Sorry, I meant to paste this link again: [https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/1027-api-unions/README.md](https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/1027-api-unions/README.md?rgh-link-date=2022-04-05T20%3A30%3A46Z)

I tried to track down the implementation of the union validations and it looks like that KEP's work fizzled out and didn't merge.

Reply:
Actually, I'm still unsure. It seems like the PR that didn't merge (https://github.com/kubernetes/kubernetes/pull/77370) maybe was just the implementation of that KEP against the the kubernetes core APIs, but perhaps we could still use the CRD extension schema called ([x-kubernetes-unions](https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/1027-api-unions/README.md#openapi))

Reply:
I tried but the `x-kubernetes-unions` does not work.  

I got this message 
> unknown field "x-kubernetes-unions" in io.k8s.apiextensions-apiserver.pkg.apis.apiextensions.v1.JSONSchemaProps;

It is not documented in the kubernetes docs, either.

Reply:
Thanks for looking into it! That's a bummer.